### PR TITLE
Fix #289 and modify some tests

### LINKED ISSUE #288: Tests return deprecation warnings
**General information:**

- emcee version: 3.0rc2
- platform: linux
- installation method (pip/conda/source/other?): source

**Problem description:**
The unit/integration tests return several deprecation warnings. Worth noting that this is not a major issue, but it does give momentary pause when seeing the tests completed with lots of warnings. Package is still usable overall though.

### Expected behavior:
Tests should return no deprecation warnings

### Actual behavior:
Tests return deprecation warnings (many duplicated).

### What have you tried so far?:
Running tests.
Uninstall/reinstall emcee (from source).
Uninstall/delete local repo/clone/reinstall (in that order).

### Minimal example:

(from terminal/console/your flavor of command prompt name)
```
py.test -v tests
```
output (duplicate warnings/actual test results removed for brevity; all tests did pass):
```
============================= test session starts ==============================
platform linux -- Python 3.6.7, pytest-3.8.2, py-1.6.0, pluggy-0.7.1 -- /home/jacaseyclyde/src/MCOrbit/venv/bin/python3
cachedir: .pytest_cache
rootdir: /home/jacaseyclyde/src/emcee, inifile:
collected 79 items                                                             

...

=============================== warnings summary ===============================
/home/jacaseyclyde/src/emcee/emcee/moves/mh.py:31: DeprecationWarning: invalid escape sequence \l
  """

/home/jacaseyclyde/src/emcee/emcee/moves/de.py:26: DeprecationWarning: invalid escape sequence \s
  """

/home/jacaseyclyde/src/emcee/emcee/utils.py:25: DeprecationWarning: 'chain' is deprecated. Use 'get_chain()' instead.
  deprecation_warning(msg)
/home/jacaseyclyde/src/emcee/emcee/utils.py:25: DeprecationWarning: 'lnprobability' is deprecated. Use 'get_log_prob()' instead.
  deprecation_warning(msg)

/home/jacaseyclyde/src/emcee/emcee/ensemble.py:259: DeprecationWarning: The 'thin' argument is deprecated. Use 'thin_by' instead.
  deprecation_warning("The 'thin' argument is deprecated. "

-- Docs: https://docs.pytest.org/en/latest/warnings.html
=================== 79 passed, 30 warnings in 134.61 seconds ===================
```

### LINKED ISSUE #289: sampler.lnprobability is transposed in emcee3?
tl;dr it looks like `sampler.lnprobability` is transposed in emcee 3 from what I would expect from emcee 2.

Sample code (copy pasta'd from the old docs):
```python
import numpy as np
import emcee

def lnprob(x, ivar):
    return -0.5 * np.sum(ivar * x ** 2)

ndim, nwalkers = 10, 100
ivar = 1. / np.random.rand(ndim)
p0 = [np.random.rand(ndim) for i in range(nwalkers)]

sampler = emcee.EnsembleSampler(nwalkers, ndim, lnprob, args=[ivar])
_ = sampler.run_mcmc(p0, 1000)

print(emcee.__version__)
print(sampler.chain.shape, sampler.lnprobability.shape)
```

## Output from emcee 2.2.1
```
2.2.1
(100, 1000, 10) (100, 1000)
```

## Output from emcee 3.0rc2
```
3.0rc2
(100, 1000, 10) (1000, 100)
```

### PR DESCRIPTION
**Pull request by OriolAbril**
There was a backwards compatibility problem with `lnprobability` (#289). I fixed it doing exactly the same as in `chain`. 

During testing I realized that the `lnprobability` test actually checked it had the wrong shape (emcee3 like), and there was no test for `get_log_prob` shape. Therefore, I added a test for `get_log_prob` shape. And while I was at it, I used `pytest.warns` to capture and check the deprecation warning of the emcee2 methods present in the tests (here I only modified the already present tests, no addition).

Eventually, I replaced the instance of `logging.warn` by `logging.warning`, which is already present and recommended in 2.7 and therefore should not be a problem for anybody.

EDIT: After seeing the results in travis, I set the strings with latex in it to raw strings to prevent the deprecation warning (the docs still build fine locally), now it should also fix #288.

To sum up: 
* fixes issue with lnprobability shape
 * modified tests to check for deprecation warnings of emcee2 parameters like chain, lnprobability and thin (instead of thin_by)
 * modified logging.warn to logging.warning
 * set strings with latex to raw strings
 * modified .coveragerc to avoid logging.warning now

I hope it helps!

### CODE REVIEW

**Comment by OriolAbril** [comment]
I have read about `appveyor`, `conda` and their issues, and I think I understand why appveyor builds are failing. However, I am not totally sure yet, and in addition, I would like to comment the possible solutions with you to check which one do you prefer.

I have found some issues with error messages similars to the ones I got, but I think this issue is not related to any of them. Instead, I believe the problem is because tqdm 4.26 is the last version present in [https://repo.continuum.io/pkgs/main/win-32/](https://repo.continuum.io/pkgs/main/win-32/) for python 3.5. Newer versions are only present for python 2.7, 3.6 or 3.7. All previous pull requests used tqdm 4.26, thus, there was no problem. Now however, conda tries to install tqdm 4.31 in python 3.5 which results in the current error. 

In travis there is an environment with python 3.5 but in linux conda does not try to install tqdm. Then, even though tqdm versions larger than 4.26 are not present in [https://repo.continuum.io/pkgs/main/linux-64/](https://repo.continuum.io/pkgs/main/linux-64/), there is no error risen.

The possible solutions that come to my mind are:

1. Install tqdm via pip before the command `conda install -q conda-build numpy scipy h5py setuptools pytest`
2. Use python 3.6 instead of 3.5
3. Pin appveyor build to tqdm 4.26

I should have no problem implementing any of the three. Whenever you have time say which one you prefer or a 4th one I haven't though about and I will implement to see if it solves the problem.

**Comment by OriolAbril** [comment]
@dfm any comments on this?

**Comment by dfm** [comment]
@OriolAbril: thanks for your patience! I'm just catching up on the backlog and I've got the builds passing again.

Would you mind opening a new pull request based on the current master that just fixes #289 and adds the lines to the test? Thanks a lot!

**Comment by OriolAbril** [comment]
I have rebased the pull request on top of current master, hope this works.

**Comment by dfm** [comment]
This is awesome. Thanks for this!!
